common/auth.go

```diff
@@ -15,6 +15,41 @@
 
 package common
 
-type AuthBackend interface {
-	
-}
\ No newline at end of file
+type Role interface {
+	Name() string
+}
+
+type role struct {
```

Comment:
Well, one is an interface, another is a struct, and, frankly, beyond names I am not 100% sure what we want to do here yet.